File: solutions/general/problem_palindrome/approach1.py
# check palindrome 
def checkPalindrome(num):
    # check number negative
    if num<0:
        return False
    elif num==0:
        return True
    
    # reverse the half of number
    x = 0
    while num>x:
        x = x*10 + num%10
        num //=10
        print(x,num)
    
    #  compare first half and reverse number
    if x==num or (x > num and x // 10 == num):
        return True
    else:
        return False
    
# Prompt the user for input
input_str = input('Enter a number: ')

# Convert the input to an integer
try:
    num = int(input_str)
except ValueError:
    print('Please enter a valid number.')
else:
    # Call the palindrome function and print the result
    is_palindrome = checkPalindrome(num)
    print(f'Is {num} a palindrome? {is_palindrome}')

File: solutions/general/problem_palindrome/approach1.js
// node.js to run the program

// Prompt the user for input
process.stdout.write("Enter a number: ");

// Handle the input
process.stdin.once("data", (data) => {
  // Convert the input to a number
  const num = parseInt(data.toString().trim(), 10);

  // Check if the input is a valid number
  if (isNaN(num)) {
    console.log("Please enter a valid number.");
  } else {
    // Call your palindrome function here
    const isPalindrome = checkPalindrome(num);
    console.log(`Is ${num} a palindrome? ${isPalindrome}`);
  }

  // Exit the process
  process.exit();
});

// check palindrome number
function checkPalindrome(num) {
  // Handle negative numbers and zero separately
  if (num < 0) return false;
  if (num === 0) return true;

  let x = 0;

  // reverse number at half of number
  while (num > x) {
    x = x * 10 + (num % 10);
    num = Math.floor(num / 10);
  }

  // compare first half and second half of number
  return num == x || (x > num && x / 10 == num) ? true : false;
}
